### PR TITLE
Use groupcall_default_join_as

### PR DESCRIPTION
**Pull request by rojserbest**

### CODE REVIEW

File: pytgcalls/pytgcalls/group_call_native.py [hunk 1]
```diff
@@ -283,6 +283,7 @@ async def get_group_call(self, group: Union[str, int, InputPeerChannel, InputPee
             raise RuntimeError(f'Can\'t get full chat by {group}')
 
         self.group_call = self.full_chat.call
+        self.join_as = self.full_chat.groupcall_default_join_as
```
Comment (MarshalX) [inline pytgcalls/pytgcalls/group_call_native.py:286]: Setting the default join as to join as in another part of the code is not obvious. Do not assign a class var. write if statement when the join as is None. If not arg and not default - my peer. If not arg and default exists - default

Comment (rojserbest) [reply pytgcalls/pytgcalls/group_call_native.py]: Can you review again?

**Review by MarshalX — APPROVED** [review]
Looks good. I will test it later

**Comment by MarshalX** [comment]
Do you try to run you PR? It doesnt works.

```python
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-46' coro=<GroupCallNative.__emit_join_payload_callback.<locals>._() done, defined at /Users/marshal/projects/tgcalls/tgcalls/pytgcalls/pytgcalls/group_call_native.py:598> exception=PeerIdInvalid('[400 PEER_ID_INVALID]: The peer id being used is invalid or not known yet. Make sure you meet the peer before interacting with it (caused by "phone.JoinGroupCall")')>
Traceback (most recent call last):
  File "/Users/marshal/projects/tgcalls/tgcalls/pytgcalls/pytgcalls/group_call_native.py", line 600, in _
    response = await self.client.send(functions.phone.JoinGroupCall(
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/methods/advanced/send.py", line 77, in send
    r = await self.session.send(
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/session/session.py", line 425, in send
    return await self._send(data, timeout=timeout)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/session/session.py", line 395, in _send
    RPCError.raise_it(result, type(data))
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/errors/rpc_error.py", line 78, in raise_it
    raise getattr(
pyrogram.errors.exceptions.bad_request_400.PeerIdInvalid: [400 PEER_ID_INVALID]: The peer id being used is invalid or not known yet. Make sure you meet the peer before interacting with it (caused by "phone.JoinGroupCall")
```

**Comment by rojserbest** [comment]
> Do you try to run you PR? It doesnt works. Its occurred when default join as exists and arg to start not passed
> 
> ```python
> ERROR:asyncio:Task exception was never retrieved
> future: <Task finished name='Task-46' coro=<GroupCallNative.__emit_join_payload_callback.<locals>._() done, defined at /Users/marshal/projects/tgcalls/tgcalls/pytgcalls/pytgcalls/group_call_native.py:598> exception=PeerIdInvalid('[400 PEER_ID_INVALID]: The peer id being used is invalid or not known yet. Make sure you meet the peer before interacting with it (caused by "phone.JoinGroupCall")')>
> Traceback (most recent call last):
>   File "/Users/marshal/projects/tgcalls/tgcalls/pytgcalls/pytgcalls/group_call_native.py", line 600, in _
>     response = await self.client.send(functions.phone.JoinGroupCall(
>   File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/methods/advanced/send.py", line 77, in send
>     r = await self.session.send(
>   File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/session/session.py", line 425, in send
>     return await self._send(data, timeout=timeout)
>   File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/session/session.py", line 395, in _send
>     RPCError.raise_it(result, type(data))
>   File "/opt/homebrew/lib/python3.9/site-packages/pyrogram/errors/rpc_error.py", line 78, in raise_it
>     raise getattr(
> pyrogram.errors.exceptions.bad_request_400.PeerIdInvalid: [400 PEER_ID_INVALID]: The peer id being used is invalid or not known yet. Make sure you meet the peer before interacting with it (caused by "phone.JoinGroupCall")
> ```

Did you interact with the peer?

**Comment by rojserbest** [comment]
I'll try it myself.

**Comment by MarshalX** [comment]
Interact? Its default_join_as_peer ;d

**Comment by rojserbest** [comment]
Weird :(

**Comment by rojserbest** [comment]
I can't install my fork. I get two pages of red text.

File: pytgcalls/pytgcalls/group_call_native.py [hunk 2]
```diff
@@ -350,7 +353,10 @@ async def start(
             raise RuntimeError('Chat without a voice chat')
 
         if join_as is None:
-            self.join_as = self.my_peer
+            if self.default_join_as:
+                self.join_as = self.default_join_as
```
Comment (MarshalX) [inline pytgcalls/pytgcalls/group_call_native.py:357]: `400 PEER_ID_INVALID` when we use default_join_as in join method as join_as param

**Review by MarshalX — CHANGES_REQUESTED** [review]

**Comment by nitanmarcel** [comment]
[**@MarshalX**](https://github.com/MarshalX) commented on [May 14, 2021, 9:38 AM GMT+3](https://github.com/MarshalX/tgcalls/pull/83#issuecomment-841044946 "2021-05-14T06:38:37Z - Replied by Github Reply Comments"):
> Interact? Its default\_join\_as\_peer ;d

It's still an user. Probably pyrogram needs to cache it first

**Pull request closed without merging**